src/read/read_ref.rs

```diff
@@ -39,6 +39,22 @@ where
         }
     }
 
+    /// Iterate over all keys in the map.
+    ///
+    /// Be careful with this function! While the iteration is ongoing, any writer that tries to
+    /// refresh will block waiting on this reader to finish.
+    pub fn keys(&self) -> impl Iterator<Item = &K> {
```

Comment:
Should I be using `impl Iterator` here or should I make a new `Keys` struct? Ditto for down below.

Reply:
So, the thing that's nice about an explicit type is that you can also implement `ExactSizeIterator`, `DoubleEndedIterator`, and `FusedIterator`. You can't do that (yet) with `impl Trait`.

Reply:
61c6642bb77fb25c2428222a2c12bd1dabaf8dc7